app/src/main/java/com/teamopensourcesmartglasses/chatgpt/AddPrompt.java

```diff
@@ -0,0 +1,268 @@
+package com.teamopensourcesmartglasses.chatgpt;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.room.Room;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Toast;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.teamopensourcesmartglasses.chatgpt.dao.PromptDao;
+import com.teamopensourcesmartglasses.chatgpt.database.PromptDatabase;
+import com.teamopensourcesmartglasses.chatgpt.entities.Prompt;
+import com.teamopensourcesmartglasses.chatgpt.events.PromptAddedEvent;
+import com.teamopensourcesmartglasses.chatgpt.events.PromptDeletedEvent;
+import com.teamopensourcesmartglasses.chatgpt.events.UserSettingsChangedEvent;
+
+import org.greenrobot.eventbus.EventBus;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class AddPrompt extends AppCompatActivity {
+
+    private PromptDao promptDao;
+    private RadioGroup radioGroupPrompts;
+    private EditText titleEditText;
+    private EditText descriptionEditText;
+    private Button closeButton;
+    private Button okButton;
+
+    private final List<Prompt> prompts = new ArrayList<>();
+    private static final int MAX_PROMPTS = 10;
+    private final HashMap<Integer, String> radioButtonDescriptions = new HashMap<>();
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
Is it better to use a fragment for this since the overall activity is a setting, i'm not sure as well